pkg/restore/restore.go

```diff
@@ -1249,13 +1249,36 @@ func (ctx *restoreContext) restoreItem(obj *unstructured.Unstructured, groupReso
 		errs.Add(namespace, err)
 		return warnings, errs
 	}
-	if isAlreadyExistsError {
-		fromCluster, err := resourceClient.Get(name, metav1.GetOptions{})
-		if err != nil {
-			ctx.log.Infof("Error retrieving cluster version of %s: %v", kube.NamespaceAndName(obj), err)
-			warnings.Add(namespace, err)
+	// check if we want to treat the error as a warning, in some cases the creation call might not get executed due to object API validations
+	// and Velero might not get the already exists error type but in reality the object already exists
+	objectExists := false
+	fromCluster, err := resourceClient.Get(name, metav1.GetOptions{})
```

Comment:
Also this first Get call needs to use a different error variable, since if this call fails, we want to add the original error to the restore, not this one. What the user needs to know is not what happened when we called Get but what happened when we called Create

Reply:
this should cover the case I suppose https://github.com/openshift/velero/pull/186/files#diff-e0498755be37b683528e2569667d67818bb676827c420aab49adc62cb6d47fe3R1263